lib/utils/resolveObject.js

```diff
@@ -124,7 +137,7 @@ function compile_value(value, stack) {
 }
 
 
-function selfRef(string, obj) {
+function selfRef(string, obj, stack) {
```

Comment:
`stack` isn't being used in this method it looks like.

Reply:
Yeah... I was thinking about removing current_context as a private external variable so that it was passed in using stack instead. What do you think is right here?